packages/js/src/preferences/helpers.ts

```diff
@@ -0,0 +1,64 @@
+import { ApiService } from '@novu/client';
+
+import type { NovuEventEmitter } from '../event-emitter';
+import type { ChannelPreferenceOverride, TODO } from '../types';
+import { PreferenceLevel } from '../types';
+import { Preference } from './preference';
+import type { UpdatePreferencesArgs } from './types';
+
+export const mapPreference = (apiPreference: {
+  template?: TODO;
+  preference: {
+    enabled: boolean;
+    channels: {
+      email?: boolean;
+      sms?: boolean;
+      in_app?: boolean;
+      chat?: boolean;
+      push?: boolean;
+    };
+    overrides?: ChannelPreferenceOverride[];
+  };
+}): Preference => {
+  const { template: workflow, preference } = apiPreference;
+  const hasWorkflow = workflow !== undefined;
+  const level = hasWorkflow ? PreferenceLevel.TEMPLATE : PreferenceLevel.GLOBAL;
+
+  return new Preference({
+    level,
+    enabled: preference.enabled,
+    channels: preference.channels,
+    workflow,
+    overrides: preference.overrides,
+  });
+};
+
+export const updatePreference = async ({
+  emitter,
+  apiService,
+  args,
+}: {
+  emitter: NovuEventEmitter;
+  apiService: ApiService;
+  args: UpdatePreferencesArgs;
+}): Promise<Preference> => {
+  const { workflowId, enabled, channel } = args;
+  try {
+    emitter.emit('preferences.update.pending', { args });
+
+    let response;
+    if (workflowId) {
+      response = await apiService.updateSubscriberPreference(workflowId, channel, enabled);
+    } else {
+      response = await apiService.updateSubscriberGlobalPreference([{ channelType: channel, enabled }]);
+    }
```

Comment:
It seems that we can unify this in the future under one endpoint.

Reply:
Correct. Let's add a comment in the code about that.

Reply:
https://linear.app/novu/issue/COM-60/consolidate-the-update-preferences-global-template-under-one-endpoint